src/tests/test_builtin_diffs.py

```diff
@@ -81,40 +82,47 @@ def backend(request):
     backend = request.param
     return backend
 
-x = np.reshape(np.arange(25), (5, 5))
+def generate_test_data(domain, n_elements=10):
+    x_arr = []
+    y_d_arr = []
+    for i in range(n_elements):
+        a = uniform(-10, 10)
+        while domain and not domain(a):
+            a = uniform(-10, 10)
```

Comment:
Maybe you could take a maximum and a minimum here instead of a domain , so we don't have to do rejection sampling.

Reply:
Modified as required.